x-pack/plugins/infra/public/components/waffle/node_context_menu.tsx

```diff
@@ -56,83 +56,69 @@ export const NodeContextMenu = injectUICapabilities(
-                  id: 'xpack.infra.nodeContextMenu.viewAPMTraces',
-                  defaultMessage: 'View {nodeType} APM traces',
-                },
-                { nodeType }
-              ),
-              href: `../app/apm#/traces?_g=()&kuery=${APM_FIELDS[nodeType]}~20~3A~20~22${
-                node.id
-              }~22`,
-              'data-test-subj': 'viewApmTracesContextMenuItem',
-            }
-          : undefined;
+      const apmTracesMenuItem = {
+        name: intl.formatMessage(
+          {
+            id: 'xpack.infra.nodeContextMenu.viewAPMTraces',
+            defaultMessage: 'View {nodeType} APM traces',
+          },
+          { nodeType }
+        ),
+        href: `../app/apm#/traces?_g=()&kuery=${APM_FIELDS[nodeType]}~20~3A~20~22${node.id}~22`,
+        'data-test-subj': 'viewApmTracesContextMenuItem',
+      };
+
+      const uptimeMenuItem = {
+        name: intl.formatMessage(
+          {
+            id: 'xpack.infra.nodeContextMenu.viewUptimeLink',
+            defaultMessage: 'View {nodeType} in Uptime',
+          },
+          { nodeType }
+        ),
+        href: createUptimeLink(options, nodeType, node),
+      };
 
-      const uptimeUrl = node.ip
-        ? {
-            name: intl.formatMessage(
-              {
-                id: 'xpack.infra.nodeContextMenu.viewUptimeLink',
-                defaultMessage: 'View {nodeType} in Uptime',
-              },
-              { nodeType }
-            ),
-            href: createUptimeLink(options, nodeType, node),
-          }
-        : undefined;
+      const showLogsLink = node.id && uiCapabilities.logs.show;
+      const showAPMTraceLink = uiCapabilities.apm && uiCapabilities.apm.show;
+      const showUptimeLink =
+        [InfraNodeType.pod, InfraNodeType.container].includes(nodeType) || node.ip;
 
       const panels: EuiContextMenuPanelDescriptor[] = [
         {
           id: 0,
           title: '',
           items: [
-            ...(nodeLogsUrl
-              ? [
-                  {
-                    name: intl.formatMessage({
-                      id: 'xpack.infra.nodeContextMenu.viewLogsName',
-                      defaultMessage: 'View logs',
-                    }),
-                    href: nodeLogsUrl,
-                    'data-test-subj': 'viewLogsContextMenuItem',
-                  },
-                ]
-              : []),
-            ...(nodeDetailUrl
-              ? [
-                  {
-                    name: intl.formatMessage({
-                      id: 'xpack.infra.nodeContextMenu.viewMetricsName',
-                      defaultMessage: 'View metrics',
-                    }),
-                    href: nodeDetailUrl,
-                  },
-                ]
-              : []),
-            ...(apmTracesUrl ? [apmTracesUrl] : []),
-            ...(uptimeUrl ? [uptimeUrl] : []),
+            ...(showLogsLink && nodeLogsMenuItem ? [nodeLogsMenuItem] : []),
+            nodeDetailMenuItem,
```

Comment:
are these top two here using the same logic as before? I thought all 4 would now be `...(showX ? [X] : [])`

Reply:
sorry... the `&& nodeLogsMenuItem` is not suppose to be there. The `nodeDetailMenuItem` is always available. The original had a check for `node.id` which is always true so it didn't make sense to add a check for something that's never null.

Reply:
ohhh makes sense :+1: